### PR TITLE
Adding AWS CodeBuild spec file.

### PR DESCRIPTION
**Pull request by MentalPower**

### CODE REVIEW

File: buildspec.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+version: 0.1
+
+phases:
+  install:
+    commands:
+      - echo Build started on `date`
+      - echo Installing required libs...
+      - npm install
```
Comment (nabeards) [inline buildspec.yaml]: 'yarn' may be better as it installs dependencies faster. not sure if that would be an issue to use instead?

**Comment by MentalPower** [comment]
Updated to use yarn and also run mocha

File: buildspec.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    commands:
+      - echo Build started on `date`
+      - echo Installing required libs...
+      - npm install -g mocha yarn
+  build:
+    commands:
+      - yarn install
+  test:
+    commands:
+      - mocha
```
Comment (nabeards) [inline buildspec.yaml:14]: use `yarn run mocha`, that will run the full suite

Comment (MentalPower) [reply buildspec.yaml:14]: That currently fails with a plethora of issues 😄 

```
prfaa-donations-backend mentalpower$ yarn run mocha
yarn run v1.1.0
$ mocha test/ --recursive
You are using the default filter for the donationOffer service. For more information about event filters see https://docs.feathersjs.com/api/events.html#event-filtering
You are using the default filter for the confirm service. For more information about event filters see https://docs.feathersjs.com/api/events.html#event-filtering


  Feathers application tests
    ✓ starts and shows the index page
    404
      ✓ shows a 404 HTML page
      ✓ shows a 404 JSON error without stack trace

  'confirmDonationOffer' hook
    1) runs the hook

  'sendEmailConfirmation' hook
    2) runs the hook

  'sendEmailThankyou' hook
    3) runs the hook

  'setDonationState' hook
    4) runs the hook

  'updateConfirmation' hook
    5) runs the hook

  'validateConfirm' hook
    ✓ runs the hook

  'validateDonationOffer' hook
    6) runs the hook

  'confirm' service
    ✓ registered the service

  'donationOffer' service
    ✓ registered the service


  6 passing (76ms)
  6 failing

  1) 'confirmDonationOffer' hook runs the hook:
     TypeError: Cannot read property 'service' of undefined
      at Promise (src/hooks/confirm-donation-offer.js:8:45)
      at Promise (<anonymous>)
      at confirmDonationOffer (src/hooks/confirm-donation-offer.js:6:16)
      at Context.it (test/hooks/confirm-donation-offer.test.js:13:16)

  2) 'sendEmailConfirmation' hook runs the hook:
     Uncaught TypeError: Cannot read property 'get' of undefined
      at fs.readFile (src/hooks/send-email-confirmation.js:19:49)
      at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:505:3)

  3) 'sendEmailThankyou' hook runs the hook:
     TypeError: Cannot read property 'donationOffer' of undefined
      at Promise (src/hooks/send-email-thankyou.js:7:29)
      at Promise (<anonymous>)
      at sendEmailThankyou (src/hooks/send-email-thankyou.js:6:16)
      at Context.it (test/hooks/send-email-thankyou.test.js:13:16)

  4) 'setDonationState' hook runs the hook:
     TypeError: Cannot read property 'service' of undefined
      at Promise (src/hooks/set-donation-create-fields.js:6:39)
      at Promise (<anonymous>)
      at setDonationState (src/hooks/set-donation-create-fields.js:3:16)
      at Context.it (test/hooks/set-donation-state.test.js:13:16)

  5) 'updateConfirmation' hook runs the hook:
     TypeError: Cannot read property 'service' of undefined
      at Promise (src/hooks/update-confirmation.js:8:39)
      at Promise (<anonymous>)
      at updateConfirmation (src/hooks/update-confirmation.js:6:16)
      at Context.it (test/hooks/update-confirmation.test.js:13:16)

  6) 'validateDonationOffer' hook runs the hook:
     TypeError: Cannot read property 'email' of undefined
      at Promise (src/hooks/validate-donation-offer.js:12:21)
      at Promise (<anonymous>)
      at validateDonationOffer (src/hooks/validate-donation-offer.js:10:16)
      at Context.it (test/hooks/validate-donation-offer.test.js:13:16)



error Command failed with exit code 6.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Comment (nabeards) [reply buildspec.yaml:14]: hm, that's strange. looks like the `app` isn't getting attached to hooks in the tests for some reason

Comment (nabeards) [reply buildspec.yaml:14]: @MentalPower can you skip running tests for now? I'll have to mockup some data for the hook tests to run properly and don't have the time to do that before tomorrow night.

Comment (MentalPower) [reply buildspec.yaml:14]: @nabeards sure, just approve the PR. Regular mocha tests run fine on my box.

**Comment by Tokynet** [comment]
Merging

**Pull request merged**